### PR TITLE
Feature/int 1105 codemod fp replace next head do not move metatags to

### PR DESCRIPTION
**Pull request by DmytroHryshyn**
- checks if tag attribute has any identifiers that are not defined in global scope
image

### CODE REVIEW

File: codemods/ts-morph/next/13/replace-next-head/index.ts [hunk 1]
```diff
@@ -138,9 +160,44 @@ export const buildContainer = <T>(initialValue: T) => {
 
 type Container<T> = ReturnType<typeof buildContainer<T>>;
 
+const isDefinedOnTheTopLevel = (
+	identifier: Identifier,
+	topLevelVariablesContainer: Container<Definition[]>,
+) => {
+	//  alternative
+	//  return definition.getFirstAncestorByKind(SyntaxKind.Block) === undefined;
+	const definitions = identifier.getDefinitionNodes() as Definition[];
```
Comment (grzpab) [inline codemods/ts-morph/next/13/replace-next-head/index.ts:169]: can this fail type-wise?

File: codemods/ts-morph/next/13/replace-next-head/index.ts [hunk 2]
```diff
@@ -138,9 +160,44 @@ export const buildContainer = <T>(initialValue: T) => {
...
+	// if identifier is within property access expression "obj.a.b", check only for its root "obj"
+	if (
+		(Node.isPropertyAccessExpression(parent) ||
+			Node.isElementAccessExpression(parent)) &&
+		identifierChildIndex !== 0
+	) {
+		return true;
+	}
+
+	// @TODO
```
Comment (grzpab) [inline codemods/ts-morph/next/13/replace-next-head/index.ts:183]: why TODO?

File: codemods/ts-morph/next/13/replace-next-head/index.ts [hunk 3]
```diff
@@ -528,19 +660,83 @@ const buildMetadataStatement = (metadataObject: Record<string, string>) => {
...
+	namedImports.forEach((namedImport) => {
+		topLevelVariablesContainer.set((prev) => [...prev, namedImport]);
+	});
+
+	const namespaceImport = importDeclaration.getNamespaceImport()?.getParent();
+
+	if (namespaceImport) {
+		topLevelVariablesContainer.set((prev) => [
+			...prev,
+			namespaceImport as NamespaceImport,
```
Comment (grzpab) [inline codemods/ts-morph/next/13/replace-next-head/index.ts:689]: is this `as` safe?

File: codemods/ts-morph/next/13/replace-next-head/index.ts [hunk 4]
```diff
@@ -528,19 +660,83 @@ const buildMetadataStatement = (metadataObject: Record<string, string>) => {
...
+		]);
+	}
+
+	const defaultImport = importDeclaration.getDefaultImport()?.getParent();
+
+	if (defaultImport) {
+		topLevelVariablesContainer.set((prev) => [
+			...prev,
+			defaultImport as ImportClause,
```
Comment (grzpab) [inline codemods/ts-morph/next/13/replace-next-head/index.ts:698]: is this `as` safe?

File: codemods/ts-morph/next/13/replace-next-head/index.ts (hunk 2 above)
Comment (DmytroHryshyn) [reply codemods/ts-morph/next/13/replace-next-head/index.ts:183]: i think there might be a way to infer globals from ts, currently i just check for a "process" identifier name

File: codemods/ts-morph/next/13/replace-next-head/index.ts (hunk 1 above)
Comment (DmytroHryshyn) [reply codemods/ts-morph/next/13/replace-next-head/index.ts:169]: it should not

**Review by grzpab — APPROVED** [review]
Let's test it thoroughly over the calcom repository for instance.

**Pull request merged**